docs/utilities/batch.md

```diff
@@ -522,14 +522,20 @@ You might want to bring custom logic to the existing `BatchProcessor` to slightl
 
 For these scenarios, you can subclass `BatchProcessor` and quickly override `success_handler` and `failure_handler` methods:
 
-* **`success_handler()`** – Keeps track of successful batch records
-* **`failure_handler()`** – Keeps track of failed batch records
+* **`success_handler()`** – is called for each successfully processed record
+* **`failure_handler()`** – is called for each failed record
+
+???+ warning
+    These functions have a common `record` argument. For backward compatibility reasons, their type is not the same:
+
+    - `success_handler`: `record` type is `dict[str, Any]`, the raw record data
+    - `failure_handler`: `record` type is a Pydantic model, either the parsed record or the  `EventSourceDataClass` record in case of validation error. Refer to [Accessing processed messages](#accessing-processed-messages) for more details
```

Comment:
```suggestion
    - `failure_handler`: `record` type can be an Event Source Data Class or your [Pydantic model](#pydantic-integration). During Pydantic validation errors, we fall back and serialize `record` to Event Source Data Class to not break the processing pipeline. Refer to [Accessing processed messages](#accessing-processed-messages) for more details.
```

Reply:
Small correction since the default brings a Event Source Data Class unless you bring a Pydantic model. When the latter happens, we fall back to Event Source Data Class upon validation errors.